### PR TITLE
Remove deprecated benchmark stuff

### PR DESCRIPTION
**Pull request by kongzii**
## Summary by CodeRabbit

- **New Features**
	- Enhanced market replication process with the introduction of `MarketType` classification.
	- Expanded market attributes to include `close_time`, `url`, and `volume` for more detailed market information.
	- Implemented field validators for better data integrity and added properties for outcome prices and probable resolutions.
	- Introduced `excluded_questions` parameter to allow filtering out specific questions from market queries.

- **Refactor**
	- Shifted references from `MarketSource` to `MarketType` across various functions to align with the new market handling approach.
	- Updated function signatures and implementations in market tooling to accommodate new parameters and attributes.

- **Documentation**
	- Updated README.md with a summary of changes related to market handling updates.

### CODE REVIEW

File: prediction_market_agent_tooling/benchmark/benchmark.py [hunk 1]
```diff
@@ -12,23 +12,26 @@
...
-        markets: t.List[Market],
+        markets: t.Sequence[AgentMarket],
         agents: t.List[AbstractBenchmarkedAgent],
         metric_fns: t.Dict[
-            str, t.Callable[[list[Prediction], list[Market]], str | float | None]
+            str,
+            t.Callable[[list[Prediction], t.Sequence[AgentMarket]], str | float | None],
         ] = {},
         cache_path: t.Optional[str] = None,
         only_cached: bool = False,
```
[the remark this answers is not in the document]
Comment (kongzii) [reply prediction_market_agent_tooling/benchmark/benchmark.py:37]: thanks!

File: prediction_market_agent_tooling/benchmark/benchmark.py [hunk 2]
```diff
@@ -136,12 +139,14 @@ def run_agents(self, enable_timing: bool = True) -> None:
...
-            def get_prediction_result(market: Market) -> tuple[str, Prediction]:
+            def get_prediction_result(
+                market: AgentMarket,
+            ) -> tuple[str, Prediction]:
                 with get_openai_callback() as cb:
                     start = time.time()
                     prediction = (
                         agent.check_and_predict(market_question=market.question)
-                        if not market.is_resolved
+                        if not market.is_resolved()
```
[the remark this answers is not in the document]
Comment (kongzii) [reply prediction_market_agent_tooling/benchmark/benchmark.py:149]: Sure, go for it.

**Comment by kongzii** [comment]
Running pytest with`RUN_PAID_TESTS=1` also went well.

File: prediction_market_agent_tooling/benchmark/utils.py [hunk 3]
```diff
@@ -1,133 +1,9 @@
...
-
-
-class CancelableMarketResolution(str, Enum):
-    YES = "yes"
-    NO = "no"
-    CANCEL = "cancel"
-    MKT = "mkt"
-
-
-class Market(BaseModel):
```
Comment (evangriffiths) [inline prediction_market_agent_tooling/benchmark/utils.py:46]: Wooo so much stuff deleted

File: prediction_market_agent_tooling/markets/agent_market.py [hunk 4]
```diff
@@ -33,17 +43,54 @@ class AgentMarket(BaseModel):
     """
 
     currency: t.ClassVar[Currency]
+    type: t.ClassVar[MarketType]
```
Comment (evangriffiths) [inline prediction_market_agent_tooling/markets/agent_market.py:46]: Doesn't look like this is used anywhere. Is it needed? If not the class definition could go back into prediction_market_agent_tooling/markets/markets.py

**Review by evangriffiths — COMMENTED** [review]
This PR is a thing of beauty ❤️. So much tidier now.

Don't think you've missed anything, except for a bit in the README.md that now needs updating:
`from prediction_market_agent_tooling.benchmark.utils import get_manifold_markets`

**Comment by kongzii** [comment]
> Don't think you've missed anything, except for a bit in the README.md that now needs updating:
from prediction_market_agent_tooling.benchmark.utils import get_manifold_markets

Fixed, thanks! Also, another fix was pushed in that commit.

File: prediction_market_agent_tooling/markets/agent_market.py (hunk 4 above)
Comment (kongzii) [reply prediction_market_agent_tooling/markets/agent_market.py:46]: True, we had `source` on the deprecated `Market` class, but in the end it's not used, so removing it. We can easily add it back if needed in the future.

**Review by evangriffiths — APPROVED** [review]

**Pull request merged**